gnovm/pkg/gnolang/garbage_collector.go

```diff
@@ -200,8 +203,18 @@ func (fv *FuncValue) VisitAssociated(vis Visitor) (stop bool) {
 		}
 	}
 
-	// Skip visiting the parent to avoid redundancy
-	// and prevent a potential cycle.
+	// Visit parent.
+	switch v := fv.Parent.(type) {
+	case nil:
+		return
+	case *Block:
+		if v != nil {
+			stop = vis(v)
+		}
+	case RefValue:
+		stop = vis(v)
+	}
```

Comment:
suggestion: maybe this to be consistent
```suggestion
	case RefValue:
		stop = vis(v)
	default:
        	panic(fmt.Sprintf("unexpected FuncValue.Parent type: %T", v))
	}
```